src/components/Loader/PeopleTable.tsx

```diff
@@ -0,0 +1,51 @@
+import { Person } from '../../types';
+import React, { useEffect, useState } from 'react';
+import { PersonLink } from './PersonLink';
+import { useParams } from 'react-router-dom';
+
+interface Props {
+  people: Person[];
+}
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const [isValidSlug, setIsValidSlug] = useState(true);
+
+  useEffect(() => {
+    if (!!slug && !people.find(p => p.slug === slug)) {
+      setIsValidSlug(false);
+    }
```

Comment:
This `useEffect` hook is being executed every time the `slug` changes. However, it's not clear why you're setting `isValidSlug` to `false` if the person is not found, and then setting it back to `true` after 2 seconds. If the person is not found, should `isValidSlug` remain `false`? Please clarify your logic here.